plugins/inputs/clickhouse/clickhouse.go

```diff
@@ -0,0 +1,355 @@
+
+  ## If "auto_discovery"" is "true" plugin tries to connect to all servers available in the cluster
+  ## with using same "user:password" described in "servers"" parameter
+  ## and get this server hostname list from "system.clusters" table
+  ## see
+  ## - https://clickhouse.tech/docs/en/operations/system_tables/#system-clusters
+  ## - https://clickhouse.tech/docs/en/operations/server_settings/settings/#server_settings_remote_servers
+  ## - https://clickhouse.tech/docs/en/operations/table_engines/distributed/
+  ## - https://clickhouse.tech/docs/en/operations/table_engines/replication/#creating-replicated-tables
+  ##    example: auto_discovery = false
+  # auto_discovery = true
+
+  ## Filter cluster names in "system.clusters" when "auto_discovery" is "true"
+  ## when this filter present then "WHERE cluster IN (...)" filter will apply
+  ## please use only full cluster names here, regexp and glob filters is not allowed
+  ## for "/etc/clickhouse-server/config.d/remote.xml"
+  ## <yandex>
+  ##  <remote_servers>
+  ##    <my-own-cluster>
+  ##        <shard>
+  ##          <replica><host>clickhouse-ru-1.local</host><port>9000</port></replica>
+  ##          <replica><host>clickhouse-ru-2.local</host><port>9000</port></replica>
+  ##        </shard>
+  ##        <shard>
+  ##          <replica><host>clickhouse-eu-1.local</host><port>9000</port></replica>
+  ##          <replica><host>clickhouse-eu-2.local</host><port>9000</port></replica>
+  ##        </shard>
+  ##    </my-onw-cluster>
+  ##  </remote_servers>
+  ##
+  ## </yandex>
+  ## 
+  ## example: cluster_include = ["my-own-cluster"]
+  # cluster_include = []
+
+  ## Filter cluster names in "system.clusters" when "auto_discovery" is "true"
+  ## when this filter present then "WHERE cluster IN (...)" filter will apply
+  ## if "cluster_include" not empty, this parameter will ignored
+  ##    example: cluster_exclude = ["my-internal-not-discovered-cluster"]
+  # cluster_exclude = []
+
+  ## Parameter which controls whether a client verifies the server's certificate chain and host name.
+  ## If http_tls_insecure_skip_verify is true, TLS accepts any certificate
+  ## presented by the server and any host name in that certificate.
+  # http_tls_insecure_skip_verify = true
+`
+
+type connect struct {
+	Cluster  string `json:"cluster"`
+	ShardNum int    `json:"shard_num"`
+	Hostname string `json:"host_name"`
+	url      *url.URL
+}
+
+func init() {
+	inputs.Add("clickhouse", func() telegraf.Input {
+		return &ClickHouse{
+			AutoDiscovery:      true,
+			InsecureSkipVerify: true,
+			Timeout:            5,
+		}
+	})
+}
+
+// ClickHouse Telegraf Input Plugin
+type ClickHouse struct {
+	Servers            []string `toml:"servers"`
+	AutoDiscovery      bool     `toml:"auto_discovery"`
+	ClusterInclude     []string `toml:"cluster_include"`
+	ClusterExclude     []string `toml:"cluster_exclude"`
+	Timeout            int      `toml:"timeout"`
```

Comment:
For timeout use a `internal.Duration` so that it can be specified as a string: `"5s"`

Reply:
fixed on https://github.com/influxdata/telegraf/pull/6441/commits/353c73b06df0fc8584c0da13a72c178060c5215c